### PR TITLE
Citizen and Veteran backend

### LINKED ISSUE #3318: Add Citizen and Veteran fields to backend
# Description

Add fields to User model for storing Citizen and Veteran status. 
Add these fields to User table, model, and factory.
Add these fields to schema (User and Applicant types _and_ inputs) so they can be queried and mutated.
Add these fields to fakeUsers.ts
Update the User.getIsProfileCompleteAttribute and User.scopeIsProfileComplete methods to check that Citizen and Veteran status have been set.

These will be used on the profile page like so:
image

# Acceptance Criteria
- [x] Citizen and Veteran status can be queried on User and Applicant queries.
- [x] Citizen and Veteran status can be set when creating/updating User/Applicant
- [x] Users are seeded with Citizen and Veteran values
- [x] Profiles are not considered complete if Citizen and Veteran values are not set
- [x] Ensure tests are updated or implemented, especially "profile complete" logic

### PR DESCRIPTION
**Pull request by vd1992**
closes #3318 
adds the two fields and assorted backend changes that go along with it

test: checkout + migrate then check adminer and play in graphQL playground

### DEPLOYMENT
Migration added

### CODE REVIEW

File: api/database/migrations/2022_08_02_175856_add_citizen_veteran_fields.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->string('citizenship')->nullable();
+            $table->boolean('veteran')->nullable();
```
Comment (petertgiles) [inline api/database/migrations/2022_08_02_175856_add_citizen_veteran_fields.php:18]: By convention, we've been naming boolean columns with a "is_" prefix.  For example, is_gov_employee, is_woman, is_indigenous.

File: api/database/helpers/ApiEnums.php [hunk 2]
```diff
@@ -251,4 +251,18 @@ public static function poolAdvertisementSecurity() : array
         ];
     }
 
+    /**
+     * Citizenship
+     */
+    const CITIZENSHIP_CITIZEN = 'CITIZEN';
+    const CITIZENSHIP_PR = 'PERMANENT_RESIDENT';
+    const CITIZENSHIP_OTHER = 'OTHER';
+    public static function citizenshipArray() : array
```
Comment (petertgiles) [inline api/database/helpers/ApiEnums.php:260]: The return type `array` as a suffix is unnecessary.  I suggest `citizenshipStatuses` to match your schema enum.

File: api/graphql/schema.graphql [hunk 3]
```diff
@@ -156,6 +158,12 @@ enum GovEmployeeType {
     INDETERMINATE
 }
 
+enum CitizenStatus {
```
Comment (petertgiles) [inline api/graphql/schema.graphql:161]: Should be `CitizenshipStatus`?

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looking great!  Could you please provide the queries and mutations you used for testing to get me started?  A few notes about naming below.

**Comment by vd1992** [comment]
@petertgiles 

```
mutation createUser($wherewhere:CreateUserInput!) {
  createUser(user:$wherewhere){
    firstName
    lastName
    citizenship
    veteran
  }
}
```

```
  "wherewhere": {
    "firstName": "aaa",
    "lastName": "bbb",
    "citizenship": "CITIZEN",
    "veteran": true
  },
```

File: api/database/migrations/2022_08_02_175856_add_citizen_veteran_fields.php (hunk 1 above)
Comment (vd1992) [reply api/database/migrations/2022_08_02_175856_add_citizen_veteran_fields.php]: done https://github.com/GCTC-NTGC/gc-digital-talent/pull/3564/commits/9c5fc787b95b250d414583691b720713a72677ab

File: api/database/helpers/ApiEnums.php (hunk 2 above)
Comment (vd1992) [reply api/database/helpers/ApiEnums.php]: https://github.com/GCTC-NTGC/gc-digital-talent/pull/3564/commits/f959c93ccef78f88438a6cf750f4d06689028e52

**Review by petertgiles — APPROVED** [review]
:100:

**Pull request merged**